.github/workflows/main.yml

```diff
@@ -42,3 +42,29 @@ jobs:
           file: ./Dockerfile
           push: true
           tags: ${{ secrets.DOCKER_HUB_USERNAME }}/prjctr02-infrastructure:latest
+
+  # For deployment on cluster we'll use self-made recipes :)
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install aws cli
+        run: |
+          which aws
+          aws --version
+          mkdir -p ~/.aws
+          touch ~/.aws/credentials
+          echo "[default]" >> ~/.aws/credentials
+          echo "aws_access_key_id = ${{ secrets.AWS_ACCESS_KEY_ID }}" >> ~/.aws/credentials
+          echo "aws_secret_access_key = ${{ secrets.AWS_SECRET_ACCESS_KEY }}" >> ~/.aws/credentials
+          cat ~/.aws/credentials
+
+      - name: Run Kubectl
+        run: |
+          mkdir -p ~/.kube
+          echo "${{ secrets.KUBE_CONFIG_DATA }}" | base64 -d >> ~/.kube/config
+          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+          chmod +x ./kubectl
+          ./kubectl version --client
+          ./kubectl apply -f .cluster/service-conf.yaml
```

Comment:
you can use kind for this: https://github.com/helm/kind-action, so don't use aws at all

Reply:
Thank you for suggestion, probably I'll use kind-actions for later homework. But for now I'm interested in playing around with my AWS account and building fully-automated pipeline from scratch.